src/api/environment.cc

```diff
@@ -138,7 +138,11 @@ void DebuggingArrayBufferAllocator::UnregisterPointerInternal(void* data,
   if (data == nullptr) return;
   auto it = allocations_.find(data);
   CHECK_NE(it, allocations_.end());
-  CHECK_EQ(it->second, size);
+  if (size > 0) {
```

Comment:
Would this be a good place to use `CHECK_IMPLIES()`?

Reply:
@cjihrig Does it make a difference? If I want to say “this check is only valid under these conditions”, then I am okay with using an `if`…

Reply:
It shouldn't make a difference. I just thought it might be a bit more concise and possibly (I'm not sure) simplify code coverage.

Reply:
Yeah, I’d prefer to keep this as it is here. I’ve noticed this because of failing tests, so I think code coverage isn’t an issue either.